Add docker compose

RCM7 opened the pull request:
Basically I added the docker-compose I usually use to spin up my game server.

This way all the mounts, port mapping, environment variables and build step is handled by docker-compose and specified in this file, reducing the size of the commands to something like:

```
docker-compose up -d
docker-compose down
docker-compose build
```

Any doubts just ask, but don't skip a look at the docker-compose documentation if you don't understand something.

Cheers

RCM7 committed:
Add docker-compose file

RCM7 committed:
Add docker-compose file

RCM7 committed:
Add docker-compose file

RCM7 committed:
Add docker-compose instructions to Readme.md

RCM7 committed:
Add docker-compose instructions to Readme.md

RCM7 committed:
Add docker-compose instructions to Readme.md

bkuhl commented:
@zopanix The Travis-CI build is failing because the environment variables haven't been configured in travis.  Can you set that up please?

zopanix commented:
Hey, the env vars are set, it should work now

bkuhl committed:
Updated to v0.13.14-experimental

bkuhl committed:
Updated to v0.13.15-experimental

RCM7 commented:
I don't believe using Travis CI is the best tool to build images automatically and put them in Docker Hub since Docker Hub themselves support github integration and automatic builds of images...

So unless Travis CI is running some tests on the image, I don't think it should be used at all.

bkuhl commented:
The value Travis-CI adds over Dockerhub is that it can tag the images in accordance with the GitHub release version number.  Dockerhub isn't capable of this kind of integration.

RCM7 commented:
Beg to disagree. It is possible and they even show you how with examples in the page where you set it up

EDIT: 
![screen shot 2016-08-08 at 21 46 02](https://cloud.githubusercontent.com/assets/2096467/17495588/903956a2-5db1-11e6-8a75-b598c35c5bb6.png)

bkuhl commented:
You're totally right, I don't remember seeing this.  This could replace the travis integration.  The full docs are here: https://docs.docker.com/docker-hub/github/

zopanix commented:
@RCM7 @bkuhl : I already tried using docker build tools but it didn't work that well. I don't have as many options as you show on your screenshot and, while everything is configured correctly, I've never seen an automated build starting on its own. That's why I've chosen travis ci to manage this.
It might I don't get how it works :p

RCM7 commented:
@zopanix I understand that. I tried to reproduce the steps in a dummy project, and while some may seem confusing, it all checked out and it works. If you'd give it a try I'm glad to help with anything :)

RCM7 commented:
Ping 😄

Commit:
Updated to v0.13.14-experimental

bkuhl committed:
Updated to v0.13.16-experimental

bkuhl committed:
Showing docker version

bkuhl commented:
Upgrading this would be on @zopanix unless he wants to give me access to the Dockerhub repo as well.

bkuhl committed:
Reverted to 0.13.15

bkuhl committed:
Updated to v0.13.16-experimental

bcrochet committed:
Add environment support to specify the saves and mods directories.

bcrochet committed:
Add environment support to specify the saves and mods directories.

zopanix commented:
Hey, sorry I've been inactive bud I'm getting installed in my new home. I moved to canada from France it will take me a week or two to be completely operationnal :)

Commit:
Updated to v0.13.20

Commit:
Updated to v0.14.0-experimental

Commit:
Updated to v0.14.1-experimental

Commit:
Updated to v0.14.2-experimental

Commit:
Updated to v0.14.3-experimental

Commit:
Updated to v0.14.4-experimental

Commit:
Updated to v0.14.5-experimental

Commit:
Updated to v0.14.6-experimental

Commit:
Updated to v0.14.7-experimental

zopanix committed:
Trying to fix master build of the travis CI

This was caused because when building on master, the travis_tag variable is empty, causing the build to fail

zopanix committed:
Changed readme

bkuhl committed:
Update README.md

bkuhl committed:
Update README.md

bkuhl committed:
Updated to v0.14.8-experimental

psykzz commented on docker-compose.yml:
```diff
@@ -0,0 +1,22 @@
+      - factorio_net
+    restart: always
+
+networks:
```
No need to specify a network as one is automatically created with `bridge` type.

psykzz commented on docker-compose.yml:
```diff
@@ -0,0 +1,22 @@
+      - ./mods:/opt/factorio/mods
+    ports:
+      - "34197:34197/udp"
+    environment:
```
Defaults for the environment are already provided in the Dockerfile, is there a need to redefine them?

psykzz reviewed (changes requested):
I suggest to remove the default network and overridden environment variables.

RCM7 committed:
Merge branch 'bcrochet-add-docker-compose' into add-docker-compose

RCM7 committed:
Remove redundant code

Fix image comment to new version of factorio

RCM7 committed:
Remove redundant code

Fix image comment to new version of factorio

RCM7 committed:
Merge branch 'bcrochet-add-docker-compose' into add-docker-compose

RCM7 committed:
Make tag in comment and README version independent

RCM7 committed:
Merge branch 'add-docker-compose' of github.com:RCM7/docker_factorio_server into add-docker-compose

RCM7 committed:
Fix leftovers of merge in README

RCM7 commented:
Sorry for the delay, I was out in vacation and then too occupied catching up with work 😄
